configmap/informed_watcher.go

```diff
@@ -163,8 +161,13 @@ func (i *InformedWatcher) addConfigMapEvent(obj interface{}) {
 	i.OnChange(configMap)
 }
 
-func (i *InformedWatcher) updateConfigMapEvent(old, new interface{}) {
-	configMap := new.(*corev1.ConfigMap)
+func (i *InformedWatcher) updateConfigMapEvent(o, n interface{}) {
+	// Ignore updates that are idempotent. We are seeing those
+	// periodically.
+	if equality.Semantic.DeepEqual(o, n) {
+		return
```

Comment:
With the resync period set to 10 hours, I'd kinda like to know why this is happening as often as it is rather than just blindly ignoring it.  If we make this a panic and look at the resulting stacktrace, what's making this happen?

If we want to ignore resyncs, then we can disable them by simply passing 0, but I'd rather understand this better as it may affect our reconcilers as well.

Reply:
I checked the implementation. And now I'll need to forget the details of the shared informers. But anyway the k8s ConfigMap client Watch returns updates which are decoded by the internals of the shared informer and passed on to us.
So we can add time as well to check how long since last update and pass along the ones which are > resync period.